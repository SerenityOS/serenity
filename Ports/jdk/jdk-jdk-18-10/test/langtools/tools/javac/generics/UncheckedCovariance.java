/*
 * @test /nodynamiccopyright/
 * @bug 4949303
 * @summary A method returning a raw type cannot override a method returning a generic type
 * @author gafter
 *
 * @compile  UncheckedCovariance.java
 * @compile/fail/ref=UncheckedCovariance.out -XDrawDiagnostics  -Xlint:unchecked -Werror  UncheckedCovariance.java
 */

class UncheckedCovariance {
    static class Box<T> { }
    static class A {
        Box<Integer> f() { return null; }
    }
    static class B extends A {
        Box f() { return null; }
    }
}
