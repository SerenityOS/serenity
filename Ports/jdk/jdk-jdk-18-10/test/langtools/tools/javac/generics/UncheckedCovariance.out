UncheckedCovariance.java:17:13: compiler.warn.override.unchecked.ret: (compiler.misc.unchecked.override: f(), UncheckedCovariance.B, f(), UncheckedCovariance.A), UncheckedCovariance.Box, UncheckedCovariance.Box<java.lang.Integer>
- compiler.err.warnings.and.werror
1 error
1 warning
